driver/src/http/mod.rs

```diff
@@ -0,0 +1,82 @@
+//! Module contains the implementation for a shared HTTP client for various
+//! driver components.
+
+use anyhow::{anyhow, Result};
+use isahc::http::{HttpTryFrom, Uri};
+use isahc::prelude::Request;
+use isahc::{HttpClientBuilder, ResponseExt};
+use serde::de::DeserializeOwned;
+use std::time::Duration;
+
+#[cfg(test)]
+lazy_static::lazy_static! {
+    /// An HTTP factory used for testing.
+    pub static ref TEST_FACTORY: HttpFactory = HttpFactory::new(Duration::from_secs(10));
```

Comment:
I'd expect a test factory to return mocked versions of HTTPClient (to set expectations without doing network calls). I guess this is a real factory used in the #[ignore] tests. Not sure if the name could capture this somehow. Would it maybe make sense to implement Default on HttpFactory?

Reply:
Maybe we can rename this to `GLOBAL_FACTORY`?. This is really intended for creating real actual HTTP clients for the ignored tests.

Reply:
Just implemented `Default`, it makes the most sense.